### PR TITLE
Convert prometheus and grafana ingress to ingressgateway

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR sets up an initial configuration for ingressgateways for both the `prometheus` and `grafana` add-ons. This is a requested feature to support easy access to the UIs for these addons.

This PR adds two new ports to the `ingressgateway`:
- 15030 (for `prometheus`)
- 15031 (for `grafana`)

The prior `ingress` spec artifacts are removed.

This PR replaces the following PRs:
- #7249
- #7127

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -211,6 +211,14 @@ gateways:
     - port: 8060
       targetPort: 8060
       name: tcp-citadel-grpc-tls
+    # Telemetry-related ports are enabled in gateway - but will only redirect if
+    # the gateway configration for the various components are enabled.
+    - port: 15030
+      targetPort: 15030
+      name: http-prometheus
```
[the remark this answers is not in the document]
Comment (rshriram) [reply install/kubernetes/helm/istio/values.yaml]: Also http2-grafana

File: install/kubernetes/helm/istio/values.yaml [hunk 2]
```diff
@@ -211,6 +211,14 @@ gateways:
...
+    - port: 15031
+      targetPort: 15031
+      name: http-grafana
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values.yaml]: http2-grafana

File: install/kubernetes/helm/istio/charts/prometheus/templates/gateway.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+metadata:
+  name: prometheus-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 15030
+      name: http-prometheus
+      protocol: HTTP
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/prometheus/templates/gateway.yaml]: Http2

**Comment by rshriram** [comment]
@douglas-reid please use this YAML instead of the ones you are adding:

```yaml
{{- if .Values.ingressgateway.enabled -}}
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: telemetry-gateway
  namespace: {{ .Release.Namespace }}
spec:
  selector:
    istio: ingressgateway
  servers:
  - port:
      number: 17777
      name: http2-telemetry
      protocol: HTTP2
    hosts:
    - "*"
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: grafana
  namespace: {{ .Release.Namespace }}
spec:
  host: grafana.{{ .Release.Namespace }}.svc.cluster.local
  trafficPolicy:
    tls:
      mode: DISABLE
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: prometheus
  namespace: {{ .Release.Namespace }}
spec:
  host: prometheus.{{ .Release.Namespace }}.svc.cluster.local
  trafficPolicy:
    tls:
      mode: DISABLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: telemetry-virtual-service
  namespace: {{ .Release.Namespace }}
spec:
  hosts:
  - "*"
  gateways:
  - telemetry-gateway
  http:
  - match:
    - port: 17777
      uri:
        prefix: /istio/grafana
    rewrite:
      uri: /
    route:
    - destination:
        host: grafana.{{ .Release.Namespace }}.svc.cluster.local
        port:
          number: 3000
    - port: 17777
      uri:
        prefix: /istio/prometheus
    rewrite:
      uri: /
    route:
    - destination:
        host: prometheus.{{ .Release.Namespace }}.svc.cluster.local
        port:
          number: 9090
---
{{- end }}
```

**Comment by douglas-reid** [comment]
Even rewriting to (what I believe to be) the intended config of:

```
spec:
    gateways:
    - telemetry-gateway
    hosts:
    - '*'
    http:
    - match:
      - port: 17777
        uri:
          prefix: /istio/grafana
      rewrite:
        uri: /
      route:
      - destination:
          host: grafana.istio-system.svc.cluster.local
          port:
            number: 3000
    - match:
      - port: 17777
        uri:
          prefix: /istio/prometheus
      rewrite:
        uri: /
      route:
      - destination:
          host: prometheus.istio-system.svc.cluster.local
          port:
            number: 9090
```

leads to all sorts of issues related to redirects and relative URLs.

Prometheus
```
[2018-07-24T04:32:34.781Z] "GET /istio/prometheus HTTP/1.1" 302 ...
[2018-07-24T04:32:34.852Z] "GET /graph HTTP/1.1" 404 NR...
```

Grafana:
```
[2018-07-24T04:33:41.819Z] "GET /istio/grafana HTTP/1.1" 200...
[2018-07-24T04:33:41.903Z] "GET /public/build/grafana.dark.css?v5.0.4 HTTP/1.1" 404 NR...
[2018-07-24T04:33:41.903Z] "GET /public/build/app.a4b7f5a163ea89ff6420.js HTTP/1.1" 404 NR...
```

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (douglas-reid) [reply install/kubernetes/helm/istio/values.yaml]: Can you add template checks inside of `values.yaml`? It doesn't seem like that is possible.

**Comment by douglas-reid** [comment]
ok. i introduced a new chart (`telemetry-gateway`) and use it to generate a single YAML for the gateway and virtual service.

File: install/kubernetes/helm/istio/values.yaml [hunk 4]
```diff
@@ -336,49 +344,32 @@ security:
...
+  prometheusEnabled: false
+
 grafana:
   enabled: false
   replicaCount: 1
   image: grafana
+  security:
+    enabled: false
+    adminUser: admin
+    adminPass: admin
```
Comment (linsun) [inline install/kubernetes/helm/istio/values.yaml]: adminPassword here?  since you used `.Values.security.adminPassword` earlier.

Comment (douglas-reid) [reply install/kubernetes/helm/istio/values.yaml]: Great catch!

**Comment by sebastien-prudhomme** [comment]
@douglas-reid charts/prometheus/templates/NOTES.txt still references .Values.ingress which break helm installation

**Pull request merged**